.github/workflows/build.yml

```diff
@@ -76,9 +74,8 @@ jobs:
           composer config --global --no-plugins allow-plugins.symfony/flex true &&
           composer global require symfony/flex
 
-      # until psalm is updated to v5 which is compatible with Symfony 7
-      - name: Remove vimeo/psalm
-        if: matrix.symfony-version == '7.0'
+      - name: Remove vimeo/psalm until it supports PHP 8.4
```

Comment:
`vimeo/psalm` version 5 is  compatible with Symfony 7, but not compatible with PHP 8.4 :) so have to keep the removal still

Reply:
Please can we keep psalm at v4 in this branch and revert the psalm-related changes? The Psalm upgrade can then happen separately and alongside fixing any issues that v5 is now finding.

Reply:
I would not update psalm if it was possible to avoid such an update.

Psalm 4 requires Symfony <5. Not having it updated in this PR would mean removing it from each CI matrix as it will simply not be able to be installed.

Reply:
But we must have been installing psalm 4 with the symfony 5 / 6 jobs until now?

https://packagist.org/packages/vimeo/psalm#4.30.0 looks like it only requires `symfony/console:  ^3.4.17 || ^4.1.6 || ^5.0 || ^6.0` unless there's a constraint hiding in one of the other dependencies?

Reply:
let me try to revert this in another PR against this PR and I will report back if it works.

Reply:
psalm was reverted. it now produces a build failure because of the outdated source code. I am not fixing that code in this PR.